c/misra/test/rules/DIR-4-9/test.c

```diff
@@ -0,0 +1,35 @@
+#define MACRO(OP, L, R) ((L)OP(R))                      // COMPLIANT
+#define MACRO2(L, R) (L + R)                            // COMPLIANT
+#define MACRO3(L, R) (L " " R " " L)                    // COMPLIANT
+#define MACRO4(x) (x + 1)                               // NON_COMPLIANT
+#define MACRO5(L, LR) (LR + 1)                          // COMPLIANT
+#define MACRO6(x) printf_custom("output = %d", test##x) // COMPLIANT
+#define MACRO7(x) #x                                    // COMPLIANT
+#define MACRO8(x) "NOP"                                 // COMPLIANT
+#define MACRO9() printf_custom("output = %d", 7)        // NON_COMPLIANT
+#define MACRO10(x)                                      // COMPLIANT
+
+const char a1[MACRO2(1, 1) + 6];
+extern printf_custom();
+int test1;
```

Comment:
Spurious definition.
```suggestion
```

Reply:
oh, its actually needed for the expansion of MACRO6, without it we get the compiler error:
`line 30: error: identifier "test1" is undefined`

Reply:
Ah, completely right. Will compile check next time.